### DISCUSSION TITLE
请问规则引擎中的SQL长度有限制么，在sql里面通过大量case when 做温度告警级别判断这种方式是否可取。

### DISCUSSION THREAD

**Discussion by lliangdd** [emqx/emqx#5709] [Q&A] [answered]
**Please describe your problem in detail, if necessary, you can upload the log file through the attachment**:
CASE 
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 36.0 and item.type=31 THEN 1
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 65.0 and item.type=31 THEN 2
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 104.0 and item.type=31 THEN 3
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData > 104.0 and item.type=31 THEN 4
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 45.0 and item.type=31 THEN 1
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 93.0 and item.type=31 THEN 2
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 120.0 and item.type=31 THEN 3
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData > 120.0 and item.type=31 THEN 4
WHEN item.temperatureData <= 65.0 and item.type=31 THEN 1
WHEN item.temperatureData <= 89.0 and item.type=31 THEN 2
WHEN item.temperatureData <= 109.0 and item.type=31 THEN 3
WHEN item.temperatureData > 109.0 and item.type=31 THEN 4
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 36.0 and item.type=32 THEN 1
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 65.0 and item.type=32 THEN 2
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 104.0 and item.type=32 THEN 3
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData > 104.0 and item.type=32 THEN 4
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 45.0 and item.type=32 THEN 1
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 93.0 and item.type=32 THEN 2
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 120.0 and item.type=32 THEN 3
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData > 120.0 and item.type=32 THEN 4
WHEN item.temperatureData <= 65.0 and item.type=32 THEN 1
WHEN item.temperatureData <= 89.0 and item.type=32 THEN 2
WHEN item.temperatureData <= 109.0 and item.type=32 THEN 3
WHEN item.temperatureData > 109.0 and item.type=32 THEN 4
WHEN xuLieId = '01' and goupId= '01' and   item.temperatureData <= 36.0 and item.type=1 THEN 1
WHEN xuLieId = '01' and goupId= '01' and   item.temperatureData <= 65.0 and item.type=1 THEN 2
WHEN xuLieId = '01' and goupId= '01' and   item.temperatureData <= 104.0 and item.type=1 THEN 3
WHEN xuLieId = '01' and goupId= '01' and   item.temperatureData > 104.0 and item.type=1 THEN 4
WHEN xuLieId = '01' and goupId= '02' and   item.temperatureData <= 45.0 and item.type=1 THEN 1
WHEN xuLieId = '01' and goupId= '02' and   item.temperatureData <= 93.0 and item.type=1 THEN 2
WHEN xuLieId = '01' and goupId= '02' and   item.temperatureData <= 120.0 and item.type=1 THEN 3
WHEN xuLieId = '01' and goupId= '02' and   item.temperatureData > 120.0 and item.type=1 THEN 4
WHEN item.temperatureData <= 65.0 and item.type=1 THEN 1
WHEN item.temperatureData <= 89.0 and item.type=1 THEN 2
WHEN item.temperatureData <= 109.0 and item.type=1 THEN 3
WHEN item.temperatureData > 109.0 and item.type=1 THEN 4
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 63.0 and item.type=11 THEN 1
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 65.0 and item.type=11 THEN 2
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 104.0 and item.type=11 THEN 3
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData > 104.0 and item.type=11 THEN 4
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 45.0 and item.type=11 THEN 1
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 93.0 and item.type=11 THEN 2
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 120.0 and item.type=11 THEN 3
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData > 120.0 and item.type=11 THEN 4
WHEN item.temperatureData <= 65.0 and item.type=11 THEN 1
WHEN item.temperatureData <= 89.0 and item.type=11 THEN 2
WHEN item.temperatureData <= 109.0 and item.type=11 THEN 3
WHEN item.temperatureData > 109.0 and item.type=11 THEN 4
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 36.0 and item.type=2 THEN 1
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 65.0 and item.type=2 THEN 2
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 104.0 and item.type=2 THEN 3
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData > 104.0 and item.type=2 THEN 4
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 45.0 and item.type=2 THEN 1
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 93.0 and item.type=2 THEN 2
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 120.0 and item.type=2 THEN 3
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData > 120.0 and item.type=2 THEN 4
WHEN item.temperatureData <= 65.0 and item.type=2 THEN 1
WHEN item.temperatureData <= 89.0 and item.type=2 THEN 2
WHEN item.temperatureData <= 109.0 and item.type=2 THEN 3
WHEN item.temperatureData > 109.0 and item.type=2 THEN 4
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 36.0 and item.type=21 THEN 1
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 65.0 and item.type=21 THEN 2
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData <= 104.0 and item.type=21 THEN 3
WHEN xuLieId = '01' and goupId= '01' and item.temperatureData > 104.0 and item.type=21 THEN 4
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 45.0 and item.type=21 THEN 1
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 93.0 and item.type=21 THEN 2
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData <= 120.0 and item.type=21 THEN 3
WHEN xuLieId = '01' and goupId= '02' and item.temperatureData > 120.0 and item.type=21 THEN 4
WHEN item.temperatureData <= 65.0 and item.type=21 THEN 1
WHEN item.temperatureData <= 89.0 and item.type=21 THEN 2
WHEN item.temperatureData <= 109.0 and item.type=21 THEN 3
WHEN item.temperatureData > 109.0 and item.type=21 THEN 4
ELSE 5
END as myAlarmType,

**Comment by terry-xiaoyu** [comment, marked as answer]
It OK. The only problem is that it's hard to debug this SQL if it is not working as expected.

**Comment by zmstone** [comment]
an other concern, it's interpreted execution, can be slow if message throughput volume is very high.
